Realm/Realm/Transaction.cs

```diff
@@ -59,6 +88,7 @@ public void Rollback()
         {
             EnsureActionFeasibility("roll back");
             _realm.SharedRealmHandle.CancelTransaction();
+            State = TransactionState.RolledBack;
             FinishTransaction();
```

Comment:
We can probably move setting the state to `FinishTransaction` and then pass the value as an argument. Not saving a whole lot of code, but it's a bit more expressive as "this is the final thing we need to do as we close out the transaction".